### PR TITLE
[Merged by Bors] - Deploy default and support custom affinities

### PR DESCRIPTION
**Pull request by razvan**
# Description

For https://github.com/stackabletech/issues/issues/323

:green_circle: CI https://ci.stackable.tech/job/airflow-operator-it-custom/70/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by sbernauer — COMMENTED** [review]
LGTM so far, but the legacy selector migration as well as unit tests are missing

File: docs/modules/airflow/pages/pod_placement.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          labelSelector:
+            matchLabels:
+              app.kubernetes.io/component: scheduler
+              app.kubernetes.io/instance: cluster_name
+              app.kubernetes.io/name: airflow
+          topologyKey: kubernetes.io/hostname
+        weight: 70
+----
+
+In the examples above `cluster-name` is the name of the Airflow custom resource that owns these Pods.
```
Comment (sbernauer) [inline docs/modules/airflow/pages/pod_placement.adoc:83]:
```suggestion
You can configure the Pod placement of the Airflow pods as described in xref:concepts:pod_placement.adoc[].

The default affinities created by the operator are:

1. Co-locate all the Airflow Pods (weight 20)
2. Distribute all Pods within the same role (worker, webserver, scheduler) (weight 70)
```

File: rust/crd/src/affinity.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        };
+
+        let expected: StackableAffinity = validate(get_affinity("airflow", &role)).unwrap();
+
+        let affinity = airflow
+            .merged_config(&role, &rolegroup_ref)
+            .unwrap()
+            .affinity;
+
+        assert_eq!(affinity, expected);
```
Comment (sbernauer) [inline rust/crd/src/affinity.rs:98]: Same as in https://github.com/stackabletech/zookeeper-operator/pull/649#discussion_r1115366099

Comment (razvan) [reply rust/crd/src/affinity.rs:149]: done.

**Review by sbernauer — APPROVED** [review]
LGTM, thanks!

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**